app/views/core/CreateAccountModal/CreateAccountModal.coffee

```diff
@@ -0,0 +1,104 @@
+  extras: Not yet implemented
+  confirmation: Not yet implemented
+
+NOTE: BasicInfoView's two children (SingleSignOn...View) inherit from it.
+This allows them to have the same form-handling logic, but different templates.
+###
+
+module.exports = class CreateAccountModal extends ModalView
+  id: 'create-account-modal'
+  template: template
+  closesOnClickOutside: false
+  retainSubviews: true
+
+  events:
+    'click .login-link': 'onClickLoginLink'
+    'click .back-to-segment-check': -> @state.set { screen: 'segment-check' }
+
+  initialize: (options={}) ->
+    classCode = utils.getQueryVariable('_cc', undefined)
+    @signupState = new State {
+      path: if classCode then 'student' else null
+      screen: if classCode then 'segment-check' else 'choose-account-type'
+      ssoUsed: null # or 'facebook', 'gplus'
+      classroom: null # or Classroom instance
+      facebookEnabled: application.facebookHandler.apiLoaded
+      gplusEnabled: application.gplusHandler.apiLoaded
+      classCode
+      birthday: new Date('') # so that birthday.getTime() is NaN
+    }
+
+    @listenTo @signupState, 'all', @render #TODO: debounce
+
+    @listenTo @insertSubView(new ChooseAccountTypeView()),
+      'choose-path': (path) ->
+        if path is 'teacher'
+          application.router.navigate('/teachers/signup', trigger: true)
+        else
+          @signupState.set { path, screen: 'segment-check' }
+
+    @listenTo @insertSubView(new SegmentCheckView({ @signupState })),
+      'choose-path': (path) -> @signupState.set { path, screen: 'segment-check' }
+      'nav-back': -> @signupState.set { path: null, screen: 'choose-account-type' }
+      'nav-forward': (screen) -> @signupState.set { screen: screen or 'basic-info' }
+
+    @insertSubView(new CoppaDenyView({ @signupState }))
+
+    @listenTo @insertSubView(new BasicInfoView({ @signupState })),
+      'sso-connect:already-in-use': -> @signupState.set { screen: 'sso-already-exists' }
+      'sso-connect:new-user': -> @signupState.set {screen: 'sso-confirm'}
+      'nav-back': -> @signupState.set { screen: 'segment-check' }
+      'signup': -> @signupState.set { screen: 'confirmation' }
+
+    @listenTo @insertSubView(new SingleSignOnAlreadyExistsView({ @signupState })),
+      'nav-back': -> @signupState.set { screen: 'basic-info' }
+
+    @listenTo @insertSubView(new SingleSignOnConfirmView({ @signupState })),
+      'nav-back': -> @signupState.set { screen: 'basic-info' }
+      'signup': -> @signupState.set { screen: 'confirmation' }
+        
+    @insertSubView(new ConfirmationView({ @signupState }))
+
+    # TODO: Switch to promises and state, rather than using defer to hackily enable buttons after render
+    application.facebookHandler.loadAPI({ success: => @signupState.set { facebookEnabled: true } unless @destroyed })
+    application.gplusHandler.loadAPI({ success: => @signupState.set { gplusEnabled: true } unless @destroyed })
```

Comment:
No, I'd still like the social handlers to switch to using Promises. Then using them can look more like BasicInfoView's `onSubmit`.